src/ssr-template-helper.js

```diff
@@ -113,34 +107,40 @@ export class SsrTemplateHelper {
    */
   buildPayload_(
     request, mustacheTemplate, opt_templates, opt_attributes = {}) {
-    const ampComponent = dict({
-      'type': this.sourceComponent_,
-      'successTemplate': {
+    const ampComponent = dict({'type': this.sourceComponent_});
+
+    const successTemplate = (opt_templates && opt_templates['successTemplate'])
+      ? opt_templates['successTemplate']./*REVIEW*/innerHTML : null;
+    if (successTemplate || mustacheTemplate) {
+      ampComponent['successTemplate'] = {
         'type': 'amp-mustache',
-        'payload': opt_templates
-          ? this.xmls_.serializeToString(opt_templates['successTemplate'])
-          : mustacheTemplate,
-      },
-      'errorTemplate': {
+        'payload': successTemplate
+          ? successTemplate : mustacheTemplate,
```

Comment:
Nit: `successTemplate || mustacheTemplate`

Reply:
Can't find that code version anymore but seems not necessary anymore, we have a 
if (ampFormSuccessTemplate || ampListTemplate) {
out of it.